partner_communication_switzerland/wizards/portal_wizard.py

```diff
@@ -9,25 +9,71 @@
 #
 ##############################################################################
 
-from odoo import api, models
+import logging
+
+from odoo import api, fields, models
+from odoo.tools.translate import _
+from odoo.exceptions import UserError
+
+
+_logger = logging.getLogger(__name__)
 
 
 class PortalWizard(models.TransientModel):
     _inherit = 'portal.wizard'
 
+    invitation_config_id = fields.Many2one('partner.communication.config')
+
+    @api.onchange('portal_id')
+    def onchange_portal_id(self):
+        # set the values of the users created in the super method
+        res = super(PortalWizard, self).onchange_portal_id()
+
+        self.user_ids.write({
+            'invitation_config_id': self.invitation_config_id.id
+        })
+
+        return res
+
+
+class PortalWizardUser(models.TransientModel):
+    _inherit = 'portal.wizard.user'
+
+    invitation_config_id = fields.Many2one('partner.communication.config')
+    uid_communication_id = fields.Many2one('partner.communication.job')
+
     @api.multi
     def action_apply(self):
-        """
-        Generate communication for User invitation
-        """
-        res = super(PortalWizard, self).action_apply()
-        comm_obj = self.env['partner.communication.job']
-        config = self.env.ref(
-            'partner_communication_switzerland.portal_welcome_config')
-        for user in self.mapped('user_ids.user_id'):
-            comm_obj.create({
-                'partner_id': user.partner_id.id,
-                'object_ids': user.id,
-                'config_id': config.id
-            })
+        res = super(PortalWizardUser, self).action_apply()
+
+        self.mapped('partner_id').signup_prepare()
+
+        if self.env.context['create_communication']:
```

Comment:
```suggestion
        if self.env.context.get('create_communication'):
```
This avoids the code execution to fail in case `create_communication` is not present in the context

Reply:
good to know, thank you